#Hypothesis Test Exercise 1

In [ ]:
import pandas as pd
import numpy as np
import scipy
from scipy import stats

In [ ]:
data=pd.read_csv('Cutlets.csv')

In [ ]:
data.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

In [ ]:
data.describe()

Unit A     Unit B
count  35.000000  35.000000
mean    7.019091   6.964297
std     0.288408   0.343401
min     6.437600   6.038000
25%     6.831500   6.753600
50%     6.943800   6.939900
75%     7.280550   7.195000
max     7.516900   7.545900

In [ ]:
# paired two-sample t-test
stats.ttest_ind(data['Unit A'], data['Unit B'])

TtestResult(statistic=0.7228688704678063, pvalue=0.4722394724599501, df=68.0)

# from the above test we got p-value is 0.47 which is greater than alpha (0.05)
# hence H0 (Null Hypothesis) is True

#Hypothesis Test Excercise 2

In [ ]:
data1=pd.read_csv('LabTAT.csv')

In [ ]:
data1.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

In [ ]:
data1.describe()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
count    120.000000    120.000000    120.000000     120.00000
mean     178.361583    178.902917    199.913250     163.68275
std       13.173594     14.957114     16.539033      15.08508
min      138.300000    140.550000    159.690000     124.06000
25%      170.335000    168.025000    188.232500     154.05000
50%      178.530000    178.870000    199.805000     164.42500
75%      186.535000    189.112500    211.332500     172.88250
max      216.390000    217.860000    238.700000     205.18000

In [ ]:
stats.f_oneway(data1["Laboratory 1"],data1["Laboratory 2"],data1["Laboratory 3"],data1["Laboratory 4"])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

# from the above test we got p-value is 2.11e-57 which is less than alpha (0.05)
# hence HA (Alternative Hypothesis) is True

#Hypothesis Test Excercise 3

In [ ]:
data2=pd.read_csv('BuyerRatio.csv')

In [ ]:
data2

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

In [ ]:
data2.describe()

East         West        North       South
count    2.000000     2.000000     2.000000    2.000000
mean   242.500000   832.500000   743.500000  410.000000
std    272.236111   976.514465   866.205807  480.832611
min     50.000000   142.000000   131.000000   70.000000
25%    146.250000   487.250000   437.250000  240.000000
50%    242.500000   832.500000   743.500000  410.000000
75%    338.750000  1177.750000  1049.750000  580.000000
max    435.000000  1523.000000  1356.000000  750.000000

In [ ]:
stats.chi2_contingency([data2["East"], data2["West"], data2["North"], data2["South"]])

Chi2ContingencyResult(statistic=1.5959455386610577, pvalue=0.6603094907091882, dof=3, expected_freq=array([[  42.76531299,  442.23468701],
       [ 146.81287862, 1518.18712138],
       [ 131.11756787, 1355.88243213],
       [  72.30424052,  747.69575948]]))

# from the above test we got p-value is 0.66 which is greater than alpha (0.05)
# hence H0 (Null Hypothesis) is True